go/master/service.go

```diff
@@ -389,3 +395,29 @@ func (s *Service) TaskFinished(taskID int, dummy *int) error {
 	}
 	return err
 }
+
+// TaskID is a struct which client uses for reports failure.
+type TaskID struct {
+	ID    int
+	Epoch int
+}
+
+// TaskFailed tells the service that a task is failed.
+func (s *Service) TaskFailed(taskID TaskID, dummy *int) error {
+	select {
+	case <-s.ready:
+	}
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	t, ok := s.taskQueues.Pending[taskID.ID]
+	if !ok {
+		err := errors.New("pending task not found")
+		log.WithFields(s.logFields()).Warningln("TaskFailed:Pending task #%v not found.", taskID)
+		return err
```

Comment:
Should we return error here? I think it's normal if that task is no longer pending (completed by other workers).